### PR TITLE
fix: Remove uses and tests for RPC write env var in ingester, router, compactor

### PR DESCRIPTION
**Pull request by carols10cents**
Extracted from https://github.com/influxdata/influxdb_iox/pull/7049.

This commit essentially makes the `INFLUXDB_IOX_RPC_MODE` env var always on.

As https://github.com/influxdata/idpe/issues/17122 is now done, and even though I'm on vacation, I wanted to start sending in PRs for the separate commits in my big deletion PR for ease of reviewing, verifying, and merging.

I'm setting this to automerge-- if you don't want that, please remove that label before approving.

If you'd rather wait until I'm back so I can be available to fix anything I break, I understand that too, and I won't submit any more til next week, but if I can get a few into the review queue this week I thought I'd try!

Also I'm just so excited to delete all this code :) :) :)

### CODE REVIEW

**Review by philjb — APPROVED** [review]
LGTM

**Comment by philjb** [comment]
But circle found test failures.

```
failures:
    query_tests2::cases::dedup_and_predicates_parquet
    query_tests2::cases::dedup_and_predicates_parquet_ingester
    query_tests2::cases::duplicates_ingester
    query_tests2::cases::duplicates_parquet
    query_tests2::cases::duplicates_parquet_many
    query_tests2::cases::gapfill
    query_tests2::cases::pushdown
    query_tests2::cases::retention
    query_tests2::cases::several_chunks
    query_tests2::cases::two_chunks
```

Looks like a formatting/naming change issue in parquet files.

**Comment by alamb** [comment]
> But circle found test failures.



I'll try and push a commit to update them

File: influxdb_iox/tests/query_tests2/cases/in/dedup_and_predicates_parquet_ingester.sql.expected [hunk 1]
```diff
@@ -126,6 +126,6 @@
 |    |                     RecordBatchesExec: batches_groups=1 batches=1 total_rows=2    |
 |    |             CoalesceBatchesExec: target_batch_size=8192    |
 |    |               FilterExec: tag@3 = A AND time@4 = 0    |
-|    |                 ParquetExec: limit=None, partitions={1 group: [[1/1/1/1/00000000-0000-0000-0000-000000000000.parquet]]}, predicate=tag@2 = A AND time@3 = 0, pruning_predicate=tag_min@0 <= A AND A <= tag_max@1 AND time_min@2 <= 0 AND 0 <= time_max@3, output_ordering=[tag@3 ASC, time@4 ASC, __chunk_order@0 ASC], projection=[__chunk_order, bar, foo, tag, time]    |
+|    |                 ParquetExec: limit=None, partitions={1 group: [[1/1/1/1-0000-0000-0000-000000000000.parquet]]}, predicate=tag@2 = A AND time@3 = 0, pruning_predicate=tag_min@0 <= A AND A <= tag_max@1 AND time_min@2 <= 0 AND 0 <= time_max@3, output_ordering=[tag@3 ASC, time@4 ASC, __chunk_order@0 ASC], projection=[__chunk_order, bar, foo, tag, time]    |
```
Comment (alamb) [inline influxdb_iox/tests/query_tests2/cases/in/dedup_and_predicates_parquet_ingester.sql.expected:129]: the filenames created in the tests are different, so I updated the plans

**Comment by alamb** [comment]
🤔 

CI still fails like:

```
failures:

---- mem::tests::test_catalog stdout ----
thread 'mem::tests::test_catalog' panicked at 'assertion failed: `None` does not match `Some(Shard { id, shard_index, .. }) if
id == TRANSITION_SHARD_ID && shard_index == TRANSITION_SHARD_INDEX`', iox_catalog/src/interface.rs:1134:9
stack backtrace
```

**Comment by carols10cents** [comment]
oooh oooh oooh sorry i have a commit for that-- removing automerge and fixing!

**Comment by carols10cents** [comment]
And another commit that sets up the transition shard in the in-memory catalog, which should fix the other test failure.

**Comment by carols10cents** [comment]
Ok you know what.... let's start with some different commits :)

**Comment by alamb** [comment]
> If you'd rather I just update the snapshots, I can do that instead, when I get back :)

Changing the normalization code sounds good to me 👍

**Pull request closed without merging**